tsconfig.json

```diff
@@ -1,16 +1,12 @@
 {
   "compilerOptions": {
     "lib": ["es2022"],
-    "moduleResolution": "NodeNext",
-    "module": "NodeNext",
+    "moduleResolution": "Bundler",
```

Comment:
Besides being a newer option, can you explain in more detail why we want this? Does this affect TS consumers of our package too?

Reply:
TS consumers are not affected by the presence of `tsconfig.json` and in general one doesn't need to ship `tsconfig.json` at all in the distributed package. The only thing that is useful for TS consumers is properly linking the generated declarations in `package.json`. 

The benefit of `Bundler` over `NodeNext` is simply flexibility in path resolution for `import types`. With `Bundler` you can use `import('.').AType` where as `NodeNext` requires `import('./index.js').AType`. Either of those work with `Bundler`. Since `tsconfig.json` and TSC is _not_ used for transpilation using `Bundler` is just a small win for `import types`. `esm-d-ts` by default uses `Bundler`. So the only benefit here is giving a directive to IDEs that will then subsequently allow the shortened import type.